pkg/virtual/apiexport/controllers/apireconciler/apiexport_apireconciler_reconcile.go

```diff
@@ -57,29 +63,66 @@ func (c *APIReconciler) reconcile(ctx context.Context, apiExport *apisv1alpha1.A
 	oldSet := c.apiSets[apiDomainKey]
 	c.mutex.RUnlock()
 
-	// collect APIResourceSchemas
-	schemaIdentity := map[string]string{}
-
-	apiResourceSchemas := make([]*apisv1alpha1.APIResourceSchema, 0, len(apiExport.Spec.LatestResourceSchemas)+len(internalapis.Schemas))
-
-	// TODO(ncdc): switch to "ThingPermissionClaim" and remove this
-	for _, schema := range internalapis.Schemas {
-		shallow := *schema
-		shallow.ClusterName = logicalcluster.From(apiExport).String()
-		apiResourceSchemas = append(apiResourceSchemas, &shallow)
+	// Get schemas and identities for base api export
+	// schemaIdentity will map the schema to the given API export that it comes from.
+	// and potentially a
+	apiResourceSchemas, schemaIdentity, err := c.getSchemasFromAPIExport(apiExport)
+	if err != nil {
+		return err
 	}
 
-	for _, schemaName := range apiExport.Spec.LatestResourceSchemas {
-		apiResourceSchema, err := c.apiResourceSchemaLister.Get(clusters.ToClusterAwareKey(logicalcluster.From(apiExport), schemaName))
-		if err != nil && !apierrors.IsNotFound(err) {
-			return err
+	for _, pc := range apiExport.Spec.PermissionClaims {
+		internal, schema := isPermissionClaimForInternalAPI(pc)
+		if internal {
+			shallow := *schema
+			shallow.ClusterName = logicalcluster.From(apiExport).String()
+			apiResourceSchemas = append(apiResourceSchemas, &shallow)
+			schemaIdentity[shallow.Name] = exportAndClaimInfo{}
+			continue
 		}
-		if apierrors.IsNotFound(err) {
-			klog.V(3).Infof("APIResourceSchema %s in APIExport %s|% not found", schemaName, apiExport.Namespace, apiExport.Name)
+		if !internal && pc.IdentityHash != "" {
+			exports, err := c.apiExportIndexer.ByIndex(indexers.IndexAPIExportByIdentity, pc.IdentityHash)
+			if err != nil {
+				return err
+			}
+			if len(exports) != 1 {
+				return fmt.Errorf("got %d api exports expected one", len(exports))
```

Comment:
this is not a helpful error when it happens

Reply:
actually expecting one is wrong. There is not guarantee that identities are unique.